source/Octopus.Tentacle.Client/Retries/RpcCallRetryHandler.cs

```diff
@@ -41,23 +43,34 @@ public RpcCallRetryHandler(TimeSpan retryTimeout, TimeoutStrategy timeoutStrateg
         {
             Exception? lastException = null;
             var started = new Stopwatch();
+            var nextSleepDuration = TimeSpan.Zero;
+            var totalRetryCount = 0;
 
             async Task OnRetryAction(Exception exception, TimeSpan sleepDuration, int retryCount, Context context)
             {
                 lastException = exception;
+                nextSleepDuration = sleepDuration;
                 var elapsedDuration = started.Elapsed;
+                var remainingRetryDuration = RetryTimeout - elapsedDuration - sleepDuration;
```

Comment:
We didn't take into account the sleep duration so often thought we would retry when we had actually run out of time